Dynavity/Dynavity/view-model/canvas/CanvasViewModel.swift

```diff
@@ -186,15 +196,19 @@ extension CanvasViewModel {
 
 // MARK: Annotation palette controls
 extension CanvasViewModel {
-    func switchAnnotationTool(_ newTool: PKTool) {
+    var isAnnotationPaletteActive: Bool {
```

Comment:
Would it be better to have a `currentMode` variable instead which stores an enum? I think it makes the code slightly more readable and more future-proof esp if we were to add more modes that are mutually exclusive

Reply:
Does this mean that we will have 2 levels of enums? i.e. one for mode, another for the annotation tools (for the annotation mode)?

Reply:
That was my original intention, but now that I think about it, we could actually merge the two:

```swift

enum CanvasMode {
  case selection
  case annotation(ToolbarView.SelectedAnnotationTool)
  ... // possibly more modes e.g. deletion where you can single tap to delete and stuff
}
```

Reply:
Storing the `PKTool` directly isn't possible since we need to distinguish between the various `InkType`s (which are strings) for `PKInkingTool`. Would it make more sense to simply use a flattened enum? i.e. treat each annotation tool as a mode in itself.

Reply:
opps, my bad.

> Would it make more sense to simply use a flattened enum? i.e. treat each annotation tool as a mode in itself.

Yep, I think this makes sense. Also, I realise we probably need to store some state for the lastSelectedPen and lastSelectedHighlighter, unless we don't mind it resetting to the default pen / highlighter everytime.

I'm not sure if it's worth refactoring given our time constraint. I think I'll just approve this PR first and we can fix this if there's time? Feel free to merge, unless you want to refactor it now

Reply:
> Also, I realise we probably need to store some state for the lastSelectedPen and lastSelectedHighlighter, unless we don't mind it resetting to the default pen / highlighter everytime.

I took a look at this and it seems that the storing of tool state should probably be implemented within `AnnotationPalette` itself. Seems very out of scope for this PR though.